core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -305,4 +305,20 @@ abstract class TaskContext extends Serializable {
 
   /** Gets local properties set upstream in the driver. */
   private[spark] def getLocalProperties: Properties
+
+
+  /** Whether the current task is allowed to interrupt. */
+  private[spark] def interruptible(): Boolean
+
+  /**
+   *  Pending the interruption request until the task is able to
+   *  interrupt after creating the resource uninterruptibly.
+   */
+  private[spark] def pendingInterrupt(threadToInterrupt: Option[Thread], reason: String): Unit
+
+  /**
+   * Creating a closeable resource uninterruptibly. A task is not allowed to interrupt in this
+   * state until the resource creation finishes.
+   */
+  private[spark] def createResourceUninterruptibly[T <: Closeable](resourceBuilder: => T): T
```

Comment:
I noticed that the description mentions a follow-up to apply this function. Could you briefly explain how this method is invoked or provide a simple example?

Reply:
Updated the comment with an example.